cli/command/container/prune.go

```diff
@@ -28,6 +29,10 @@ func NewPruneCommand(dockerCli command.Cli) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			spaceReclaimed, output, err := runPrune(cmd.Context(), dockerCli, options)
 			if err != nil {
+				if errdefs.IsCancelled(err) {
+					fmt.Fprintln(dockerCli.Out(), output)
```

Comment:
What does `output` contain in this situation? Is this a "partial prune" that is printed? If that's the case, should use instead handle it as "normal", and print both `output` and `spaceReclaimed` ?

Something like;

```go
if err != nil && !errdefs.IsCancelled(err) {
	return nil
}
if output != "" {
	fmt.Fprintln(dockerCli.Out(), output)
}
fmt.Fprintln(dockerCli.Out(), "Total reclaimed space:", units.HumanSize(float64(spaceReclaimed)))
return nil
```

Perhaps we need a check for the `spaceReclaimed` to only print if we did reclaim (not sure?).

Reply:
On a side-note (I should open a ticket for that); `docker system prune` and `docker builder prune` are really awkward currently, as they don't print any progress until the prune finishes.

We should improve that;

- print prune progress as we go (instead of dump everything at the end)
- consider if we want `build cache` entries to be printed individually (their output is not very useful to the user), and instead possibly print a progress-bar or something along those lines)

Reply:
I'm essentially reverting the changes from this commit https://github.com/docker/cli/commit/69e0f53a032a85c79e456724df0625eb51cbb16e. 

Propagating the error up to where the cobra root command is executed does not necessarily mean we will output the error to the user. Here I've introduced logic to check which error is returned (in this case `errdefs.Cancelled`). 
See this change here https://github.com/docker/cli/blob/9bc91a81236b182e0dadb91e0d5d8d223092ab9e/cmd/docker/docker.go#L47-L49

RE: space reclaimed output, yes we should fix that so we are really sure space was reclaimed when printing vs just always printing it when no error occurred.